man/faq/developer/reference-compatibility.Rmd

```diff
@@ -37,6 +37,10 @@ Coercion errors are the main source of incompatibility with vctrs. See the [howt
 
 vctrs can achieve all necessary operations on vectors with the four primitives listed boved. However there are many higher level operations. The most important ones implement fallbacks to base generics for maximum compatibility with existing classes.
 
+- We have started to implement a fallback mechanism in [vec_cast_common()]. This function normally calls [vec_ptype2()] and [vec_cast()] to normalise a set of inputs to the same type. When the inputs do not implement the coercion generics but they all inherit from the same base class, we do not attempt any coercion and leave the inputs as is.
+
+  This fallback should be safe to use before operations like [vec_match()] or [vec_rbind()]. It is currently internal.
```

Comment:
Do you want to update these comments in light of your github comment that states this this won't work?